model_hub/model_hub/huggingface/_arg_parser.py

```diff
@@ -0,0 +1,253 @@
+from attrdict import AttrDict
+import dataclasses
+from dataclasses import dataclass, field, asdict
+from typing import Dict, Optional, Tuple
+
+
+@dataclass
+class DatasetKwargs:
+    dataset_name: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Path argument to pass to HuggingFace datasets's load_dataset.  "
+            "Can be a dataset identifier in HuggingFace Datasets Hub or a local path to processing script."
+        },
+    )
+    dataset_config_name: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "The name of the dataset configuration to pass to HuggingFace dataset's load_dataset."
+        },
+    )
+    validation_split_percentage: Optional[float] = field(
+        default=None,
+        metadata={
+            "help": "This is used to create a validation split from the training data when "
+            "a dataset does not have a predefined validation split."
+        },
+    )
+    train_file: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Path to training data.  This will be used if a dataset_name is not provided."
+        },
+    )
+    validation_file: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Path to validation data.  This will be used if a dataset_name is not provided."
+        },
+    )
+
+
+@dataclass
+class ConfigKwargs:
+    pretrained_model_name_or_path: str = field(default=None)
+    cache_dir: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Where do you want to store the pretrained models downloaded from huggingface.co"
+        },
+    )
+    revision: Optional[str] = field(
```

Comment:
Can we title revision something else -- maybe `model_version`? This name is not apparent to me as to what it means.

Reply:
this is the name used by HF so i'm hesitant to change it.